pkg/status/deployment_test.go

```diff
@@ -17,56 +16,65 @@ import (
 func TestGetDeploymentStatusConditions(t *testing.T) {
 
 	t.Run("test deployment status conditions", func(t *testing.T) {
+
+		t.Run("deployment ready", func(t *testing.T) {
+			fakeClient := test.NewFakeClient(t, fakeDeploymentReady())
+			conditions := GetDeploymentStatusConditions(fakeClient, "test-deployment", test.HostOperatorNs)
+			err := ValidateComponentConditionReady(conditions...)
+			require.NoError(t, err)
+
+			expected := toolchainv1alpha1.Condition{
+				Type:    toolchainv1alpha1.ConditionReady,
+				Status:  corev1.ConditionTrue,
+				Reason:  "DeploymentReady",
+				Message: "",
+			}
+			test.AssertConditionsMatchAndRecentTimestamps(t, conditions, expected)
+		})
+
 		t.Run("deployment does not exist", func(t *testing.T) {
 			fakeClient := test.NewFakeClient(t)
-			conditions, err := GetDeploymentStatusConditions(fakeClient, "test-deployment", test.HostOperatorNs)
+			conditions := GetDeploymentStatusConditions(fakeClient, "test-deployment", test.HostOperatorNs)
+			err := ValidateComponentConditionReady(conditions...)
 			require.Error(t, err)
```

Comment:
I think instead of indirectly testing ValidateComponentConditionReady in these tests we could just create dedicated tests for ValidateComponentConditionReady() in a separate componentconditions_test.go file with full coverage.

Reply:
I added the tests but I think it's still useful to test the error here as well so I'll keep these checks here if that's okay.